packages/nx/src/utils/path.ts

```diff
@@ -17,3 +17,28 @@ export function normalizePath(osSpecificPath: string): string {
 export function joinPathFragments(...fragments: string[]): string {
   return normalizePath(path.join(...fragments));
 }
+
+/**
+ * Detect workspace scope from the package.json name
+ * @param packageName
+ * @returns
+ */
+export function detectWorkspaceScope(packageName: string): string {
+  if (!packageName) return '';
```

Comment:
This is pedantic, but could be written succinctly as:

```
return packageName.startsWith('@') ? packageName.substring(1).split('/')[0] : '';
```

Maybe `packageName?.startsWith` if there is a chance it can be undefined/null.